File: src/main/java/org/kane/domain/DTO/entityDTO/daily_record/DiaryRecordShowDTO.java
package org.kane.domain.DTO.entityDTO.daily_record;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kane.domain.DTO.entityDTO.meal.MealShowDTO;
import org.kane.domain.DTO.entityDTO.meal_item.TotalMealShowDTO;
import org.kane.domain.DTO.entityDTO.sport_activity.CalorieConsumptionShowDTO;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DiaryRecordShowDTO {
    LocalDate date;
    List<MealShowDTO> meals;
    TotalMealShowDTO total;
    CalorieConsumptionShowDTO calorieConsumption;
}

File: src/app/DTO/entity_dto/diary/diary_record/diary-record-show.dto.ts
import type { CalendarDate } from '../../../types';
import MealShowDTO from '../meal/meal-show.dto';
import TotalMealShowDTO from '../meal_item/total-meal-show.dto';
import CalorieConsumptionShowDTO from '../sport_activity/calorie-consumption-show.dto';

export default interface DiaryRecordShowDTO {
  date: CalendarDate;
  meals: MealShowDTO[]
  total: TotalMealShowDTO;
  calorieConsumption: CalorieConsumptionShowDTO
}
